src/appleseed-max-impl/oslutils.cpp

```diff
@@ -41,16 +41,118 @@
 
 // 3ds Max Headers.
 #include <bitmap.h>
+#include <imtl.h>
+#include <maxapi.h>
 #include <stdmat.h>
 
 namespace asf = foundation;
 namespace asr = renderer;
 
+namespace
+{
+    asr::ParamArray get_uv_params(Texmap* texmap)
+    {
+        asr::ParamArray uv_params;
+
+        if (!is_bitmap_texture(texmap))
+            return uv_params;
+
+        UVGen* uv_gen = texmap->GetTheUVGen();
+        if (!uv_gen || !uv_gen->IsStdUVGen())
+            return uv_params;
+
+        StdUVGen* std_uv = static_cast<StdUVGen*>(uv_gen);
+        auto time = GetCOREInterface()->GetTime();
+
+        // Assume that all the textures are in texture mode texmap->MapSlotType() == MAPSLOT_TEXTURE.
+        // Assume that static_cast<StdUVGen*>(uv_gen)->GetCoordMapping() == UVMAP_EXPLICIT.
+        float u_tiling = std_uv->GetUScl(time);
+        float v_tiling = std_uv->GetVScl(time);
+        float u_offset = std_uv->GetUOffs(time);
+        float v_offset = std_uv->GetVOffs(time);
+        float w_rotation = std_uv->GetWAng(time);
+        int tiling = std_uv->GetTextureTiling();
+
+        if (tiling & U_WRAP)
+            uv_params.insert("in_wrapU", fmt_osl_expr(1));
+        else if (tiling & U_MIRROR)
+            uv_params.insert("in_mirrorU", fmt_osl_expr(1));
+
+        if (tiling & V_WRAP)
+            uv_params.insert("in_wrapV", fmt_osl_expr(1));
+        else if (tiling & V_MIRROR)
+            uv_params.insert("in_mirrorV", fmt_osl_expr(1));
+
+        uv_params.insert("in_offsetU", fmt_osl_expr(u_offset));
+        uv_params.insert("in_offsetV", fmt_osl_expr(v_offset));
+
+        uv_params.insert("in_tilingU", fmt_osl_expr(u_tiling));
+        uv_params.insert("in_tilingV", fmt_osl_expr(v_tiling));
+
+        uv_params.insert("in_rotateW", fmt_osl_expr(w_rotation));
+
+        // Access BMTex paramaters which are only accessible in a weird way.
```

Comment:
*parameters*

Reply:
What's the weird way of accessing them?

Reply:
It looked weird to me to not have get/set methods. Changed to something neutral, but I can remove it completely if you think it's not needed.